src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -68399,6 +68399,105 @@ static void F(object? x, object y)
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "u.y").WithLocation(14, 9));
         }
 
+        [Fact]
+        [WorkItem(46206, "https://github.com/dotnet/roslyn/issues/46206")]
+        public void Tuple_Assignment_25()
+        {
+            var source =
+@"using System;
+class Program
+{
+    static void F(object x, string y)
+    {
+        (object, string?) t = new ValueTuple<object?, string>(item2: """", item1: null) { Item1 = x };
+        t.Item1.ToString();
+        t.Item2.ToString();
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                    // (6,31): warning CS8619: Nullability of reference types in value of type '(object?, string)' doesn't match target type '(object, string?)'.
+                    //         (object, string?) t = new ValueTuple<object?, string>(item2: "", item1: null) { Item1 = x };
+                    Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, @"new ValueTuple<object?, string>(item2: """", item1: null) { Item1 = x }").WithArguments("(object?, string)", "(object, string?)").WithLocation(6, 31));
+            comp.VerifyTypes();
+        }
+
+        [Fact]
+        [WorkItem(46206, "https://github.com/dotnet/roslyn/issues/46206")]
+        public void Tuple_Assignment_26()
+        {
+            var source =
+@"using System;
+class Program
+{
+    static void F(bool b, object x, object? y)
+    {
+        var t = new ValueTuple<object?, object, object?, object, object?, ValueTuple<object, object?>, object, ValueTuple<object, object?, object>>(
+            x,
+            x,
+            x,
+            b ? y : null!, // 1
+            y,
+            new ValueTuple<object, object?>(b ? y : null!, x), // 2,
+            rest: new ValueTuple<object, object?, object>(x, b ? y : null!, b ? y : null!),
+            item7: x)/*T:(object?, object!, object?, object!, object?, (object!, object?), object!, object!, object?, object!)*/; // 3
```

Comment:
>x [](start = 19, length = 1)

Consider making this erroneously nullable, to ensure that the out-of-orderness is being tested.